### PR TITLE
Big 227 improve ex with column mismatch

### PR DESCRIPTION
**Pull request by romanbracinik**
SAPI Testy https://github.com/keboola/storage-api-php-client/pull/1247
Connection zatial len TMP https://github.com/keboola/connection/pull/4790 po mergi tu releasnem a requirnem v connection a dam na review

Da sa to tam pripadne vyskusat

### CODE REVIEW

File: tests/functional/UseCase/Bucket/ShareLinkBucketTest.php [hunk 1]
```diff
@@ -384,6 +384,11 @@ public function testShareUnshareLinkedBucket(): void
         $targetDataset = $targetProjectBqClient->dataset($linkedBucketSchemaName);
         $this->assertTrue($targetDataset->exists());
         $testTableBefore = $targetDataset->table(self::TESTTABLE_AFTER_NAME);
-        $this->assertTrue($testTableBefore->exists());
+
+        // after unshare the table is not available
+        // in connection you can't just unshare a bucket that is lined up first so this is an edge case
+        // handled in connection
+        $this->expectException(BadRequestException::class);
```
Comment (romanbracinik) [inline tests/functional/UseCase/Bucket/ShareLinkBucketTest.php:391]: Zas to zacalo padat, a ked som vratil to co tam bolo povodne tak to zacalo fungovat.

File: src/Handler/Table/Import/ImportTableFromTableHandler.php [hunk 2]
```diff
@@ -205,11 +206,15 @@ private function import(
             // this will skip moving data to stage table
             // this is used on full load into workspace where data are deduplicated already
             $toStageImporter = new ToStageImporter($bqClient);
-            $importState = $toStageImporter->importToStagingTable(
-                $source,
-                $destinationDefinition,
-                $importOptions,
-            );
+            try {
+                $importState = $toStageImporter->importToStagingTable(
```
Comment (romanbracinik) [inline src/Handler/Table/Import/ImportTableFromTableHandler.php:211]: fullload

File: src/Handler/Table/Import/ImportTableFromTableHandler.php [hunk 3]
```diff
@@ -222,6 +227,8 @@ private function import(
                 $stagingTable,
                 $importOptions,
             );
+        } catch (ColumnsMismatchException $e) {
```
Comment (romanbracinik) [inline src/Handler/Table/Import/ImportTableFromTableHandler.php:231]: Increment

**Comment by zajca** [comment]
Není název té exception fail? Vlastně je tam rozdíl Columns->Column klidně bych ten název udělal stejný a udělal class alias je to hrozně zmatečné, chvilku sem si myslel, že vyhazuješ stejnů exception :D

**Comment by romanbracinik** [comment]
> Není název té exception fail? Vlastně je tam rozdíl Columns->Column klidně bych ten název udělal stejný a udělal class alias je to hrozně zmatečné, chvilku sem si myslel, že vyhazuješ stejnů exception :D

https://github.com/keboola/php-storage-driver-bigquery/pull/106/commits/bbf136e2835a4ed39cf6954b17e0b2df12193c3c

File: tests/functional/UseCase/Bucket/ShareLinkBucketTest.php [hunk 4]
```diff
@@ -384,6 +384,11 @@ public function testShareUnshareLinkedBucket(): void
         $targetDataset = $targetProjectBqClient->dataset($linkedBucketSchemaName);
         $this->assertTrue($targetDataset->exists());
         $testTableBefore = $targetDataset->table(self::TESTTABLE_AFTER_NAME);
-        $this->assertTrue($testTableBefore->exists());
+
+        // after unshare the table is not available
+        // in connection you can't just unshare a bucket that is lined up first so this is an edge case
```
Comment (tomasfejfar) [inline tests/functional/UseCase/Bucket/ShareLinkBucketTest.php:389]: lined up?

File: tests/functional/UseCase/Bucket/ShareLinkBucketTest.php (hunk 1 above)
Comment (tomasfejfar) [reply tests/functional/UseCase/Bucket/ShareLinkBucketTest.php:391]: Tomu nějak nerozumím :(

File: tests/functional/UseCase/Table/Import/FromTable/ImportTableFromTableTest.php [hunk 5]
```diff
@@ -691,4 +695,128 @@ public function testLoadNullsToRequiredFields(): void
...
+        $qb = new BigqueryTableQueryBuilder();
+        $sql = $qb->getCreateTableCommand(
+            $tableSourceDef->getSchemaName(),
+            $tableSourceDef->getTableName(),
+            $tableSourceDef->getColumnsDefinitions(),
+            $tableSourceDef->getPrimaryKeysNames(),
+        );
+        $bqClient->runQuery($bqClient->query($sql));
+        $insert = [];
+        // data in source table have nulls -> should fail, because we are trying to fit null value in to required field
```
Comment (tomasfejfar) [inline tests/functional/UseCase/Table/Import/FromTable/ImportTableFromTableTest.php]: Žádné nully v tom insertu nejsou, ne?

Comment (tomasfejfar) [reply tests/functional/UseCase/Table/Import/FromTable/ImportTableFromTableTest.php]: Jinak ten test mi přijde OK (byť možná vkládat tam víc řádků je zbytečné - mohls tam napsat celou query jako string a vložit jeden řádek).

**Review by tomasfejfar — APPROVED** [review]
Funguje, ty poznámky jsou asi jen formální...
